lnwallet/channel.go

```diff
@@ -6196,8 +6205,16 @@ func NewLocalForceCloseSummary(chanState *channeldb.OpenChannel,
 		return nil, err
 	}
 
+	// We use a defaultMaxDeadline here as a LocalForceCloseSummary is
+	// created under two cases,
+	//  - when the chain watcher detects a spend of the commit tx, hence
+	//    it's confirmed, thus we have no deadline for the commit tx.
+	//  - when the channel arbitrator decides to perform a local force
+	//    close, the state would transit from StateBroadcastCommit to
```

Comment:
since this is the only case (?) the deadline is really needed to be set, can instead move the dealine calculation there (I think the channel arbitrator is already checking HTLC timeouts when deciding to go to chain).

We could also pass the deadline in as a variadic functional argument, so we would only have to pass it in where it was really needed.

Reply:
Yeah if we only need deadline for local force close, which I believe is the case atm, then I don't think here's the best place to put down the logic. The whole `lnwallet` can be unaware of the deadline, and put the calculation in `ChannelArbitrator`, in [`sweepAnchors`](https://github.com/lightningnetwork/lnd/blob/master/contractcourt/channel_arbitrator.go#L1090). Another reason is that, it seems we are remaking the anchor resolutions in `sweepAnchors` anyway, what's made here will be overwrite when we hit `case StateCommitmentBroadcasted`.